integrators/embedded_explicit_runge_kutta_nyström_integrator_body.hpp

```diff
@@ -239,19 +250,55 @@ Status EmbeddedExplicitRungeKuttaNyströmIntegrator<Position,
       q_hat[k].Increment(Δq_hat[k]);
       v_hat[k].Increment(Δv_hat[k]);
     }
-    problem.append_state(current_state);
+    append_state(current_state);
     ++step_count;
     if (step_count == adaptive_step_size.max_steps && !at_end) {
       return Status(termination_condition::ReachedMaximalStepCount,
                     "Reached maximum step count " +
                         std::to_string(adaptive_step_size.max_steps) +
                         " at time " + DebugString(t.value) +
-                        "; problem t_final is " + DebugString(problem.t_final) +
+                        "; problem t_final is " + DebugString(t_final) +
                         ".");
     }
   }
+  // The resolution is restartable from the last non-truncated state.
+  if (final_state) {
```

Comment:
I think by this point `final_state` is always engaged, thus no need for the conditional; if that's not the case, this code might be incorrect, since setting `final_state` can be followed by picking a shorter time step instead if `tolerance_to_error_ratio > 1`.